string ip_addr
uint32 dst_port
uint32 src_port